jsexception.access.denied=You do not have permission to take the requested action. Log in as a permitted user if you can.
jsexception.any={0}

jsexception.bean.has.no.method=The bean with the name {0} does not have method {1}.
jsexception.bean.name.not.found.for.interface=No bean name was found for interface {0}.
jsexception.bean.no.name=A bean with the name {0} cannot be found.
jsexception.bean.not.a.ReportDataSourceService=The {0} bean is not a ReportDataSourceService.

jsexception.call.not.supported=The call is not supported by resource lookup objects.
jsexception.connection.type.not.suported=The connection type is not supported.
jsexception.connection.uri.not.specified=The connection URI is not specified.
jsexception.could.not.find.resource.with.uri=The resource with URI {0} cannot be found. 
jsexception.could.not.find.content.resource.with.uri=The content resource with URI {0} cannot be found.
jsexception.unexpected.resource.type.at.uri=Unexpected type ({0}) of the resource with URI {1} 

jsexception.default.message=An unexplained error occurred. Please contact your system administrator.

jsexception.error.creating.connection=An error occurred while creating the connection. Try again.
jsexception.parameter.has.multiple.values=Parameter {0} has multiple values.

jsexception.parameter.has.no.value=Parameter {0} has no value.


jsexception.failed.to.create.connection=The connection could not be made.
jsexception.failed.to.save.connection=The connection was not saved.
jsexception.failed.to.save.connection.changes=The connection changes were not saved.
jsexception.file.resource.is.reference=File resource {0} is a reference and has no data.
jsexception.file.resource.no.match.type=The declared type of resource ({0}) does not match the actual type of the reference ({1}).
jsexception.folder.already.exists=A folder with the name {0} already exists. Choose a different name.
jsexception.folder.details.not.found=Folder details for folder {0} were not found in the import information. 
jsexception.folder.not.found=The folder {0} cannot be found. 
jsexception.folder.not.found.at=The folder is not at {0}.
jsexception.folder.duplicate.name=There is already a folder named {0} in {1}.
jsexception.folder.duplicate.label=Duplicate folder label {0} in folder {1}.
jsexception.folder.too.long.uri=The URI of the folder is too long. The maximum length is {0} characters.

jsexception.got.null.schema.source.wrapper=The subflow returned a null schema source wrapper.
jsexception.got.null.access.grant.source.wrapper=The subflow returned a null access grant source wrapper.
jsexception.got.null.connection.back=The subflow returned a null connection.

jsexception.implementation.class.does.not.implement=Implementation class {0} does not implement {1}.
jsexception.implementation.class.not.found=An implementation class for {0} cannot be found.
jsexception.import.directory.not.found=The import directory {0} cannot be found.
jsexception.import.module.not.found=The import module {0} cannot be found.
jsexception.import.multiple.values.for.single.parameter=Multiple parameter values were found for the single-value parameter {0} of report {1}.
jsexception.input.stream.exhausted=The input stream was exhausted before reading {0} bytes.
jsexception.invalid.bean.datasource=The bean data source {0} is invalid.
jsexception.invalid.class.attribute.for.model.tag=The class attribute for model tag, resource {0}, is invalid.
jsexception.invalid.jdbc.datasource=The JDBC data source {0} is invalid.
jsexception.invalid.aws.datasource=The AWS data source {0} is invalid.
jsexception.invalid.jndi.jdbc.datasource=The JNDI JDBC data source {0} is invalid.
jsexception.invalid.olap.datasource=The OLAP data source {0} is invalid.
jsexception.invalid.resource.reference.source=The resource reference source {0} is invalid.
jsexception.invalid.session=The current session is invalid. Make sure the server is running and try logging in again.
jsexception.invalid.uri=The URI {0} is invalid.

jsexception.jar.entry.not.found=A JAR entry for URL {0} cannot be found.
jsexception.jar.malformed.url=The JAR URL {0} has incorrect syntax.
jsexception.jasperprint.not.found=The JasperPrint object {0} cannot be found.

jsexception.job.has.more.than.one.trigger=Job {0} has more than one trigger.
jsexception.job.no.versions.match=Client version {0} is not the latest version ({1}).
jsexception.job.unknown.calendar.trigger.days.type=Unknown calendar trigger days type {0}.
jsexception.job.unknown.interval.unit=Unknown report job interval unit {0}.
jsexception.job.unknown.start.type=Unknown report job start type {0}.
jsexception.job.unknown.trigger.bean.class=Unknown report job trigger bean class {0}.
jsexception.job.unknown.trigger.type=Unknown report job trigger type {0}.

jsexception.missing.olap.connection=The OLAP connection is missing.
jsexception.missing.olap.connection.reference=The OLAP connection reference is missing.
jsexception.mondrian.jdbc.datasource.has.an.invalid.type=The Mondrian connection JDBC data source is an invalid type ({0}).

jsexception.no.connection.selected=No connection is selected.
jsexception.no.export.folder.specified=No export folder was specified.
jsexception.no.export.zip.file.specified=No export ZIP file was specified.
jsexception.no.import.folder.specified=No import folder was specified.
jsexception.no.import.zip.file.specified=No import ZIP file was specified.
jsexception.no.input.parameter.specified=No input parameter was specified.
jsexception.no.olap.client.connection=No OLAP client connection can be found.
jsexception.no.secure.olap.client.connection=No Secure OLAP client connection can be found.
jsexception.no.olap.model.created.for=No OLAP Model was created for {0}.
jsexception.no.olap.model.name=No OLAP Model was specified.
jsexception.no.olap.model.retrieved=No OLAP Model could be retrieved.
jsexception.no.output.parameter.specified=No output parameter was specified.
jsexception.no.persistent.class.mapped.to=No persistent class is mapped to {0}.
jsexception.no.resource.data.provider.found=No resource data provider was found for ID {0}.
jsexception.no.such.object.permission=The specified object permission does not exist.
jsexception.no.such.role=There is no role {0}.
jsexception.no.such.user=There is no user {0}.
jsexception.no.uri.or.recipient.given=No URI or recipient was specified.
jsexception.no.values.to.enumerate=No values were specified to enumerate.
jsexception.no.values.to.enumerate={0} is not a valid resource URI.
jsexception.null.uri=Null URI.

jsexception.output.stream.already.closed=The output stream is already closed.

jsexception.property.not.a.list=Property {0} of bean {1} is not a list property.
jsexception.property.not.a.map=Property {0} of bean {1} is not a map property.
jsexception.property.not.a.text=Property {0} of bean {1} is not a text property.

jsexception.query.datasource.has.an.invalid.type=The query data source has an invalid type ({0}).

jsexception.report.job.not.found=A report job with ID {0} cannot be found.
jsexception.report.resource.already.exists.no.overwrite=Output resource {0} already exists and the job overwrite flag is not set.
jsexception.report.no.content.resource=Resource {0} is not a content resource.
jsexception.report.unit.datasource.has.an.invalid.type=The report unit data source has an invalid type ({0}).
jsexception.report.unknown.output.format=Report output format {0} is not a known format.
jsexception.repository.uri.neither.report.nor.folder=The repository URI {0} is neither a report unit nor a folder.
jsexception.resource.already.exists=Resource {0} already exists. Choose a different name.
jsexception.resource.copyTo.not.implemented=Resource copyTo is not implemented.
jsexception.resource.copyFrom.not.implemented=Resource copyFrom is not implemented.
jsexception.resource.does.not.exist=Resource {0} does not exist.
jsexception.resource.no.match.versions={0}: client version {1} does not match the repository version ({2}).
jsexception.resource.of.type.not.found=Resource {0} of type {1} cannot be found.
jsexception.resource.not.report.unit=Resource {0} is not a report unit.
jsexception.resourcelookup.not.a.resource=ResourceLookup is not a resource: {0}.
jsexception.folder.delete.root=The root folder cannot be deleted.
jsexception.report.template.not.found=The repository URI {0} is not template.
jsexception.report.template.uri.not.specified=The template URI is not specified.

jsexception.selected.report.not.adhoc=The selected report is not an Ad Hoc report. Regular report units are not supported.
jsexception.session.timeout.occurred=The OLAP session timed out.

jsexception.too.many.data.rows=The Excel exporter failed because of too many data rows. You can change your exporter settings to one page per sheet.

jsexception.unable.to.complete.request=The system was unable to complete you request. Your conversation may have ended, expired, or been invalidated.
jsexception.undefined.file.type=The file type  is not recognized.
jsexception.unknown.connection.type=The connection type is not recognized.
jsexception.unknown.parameter.type.for.multiple.value.input=The parameter type {0} is not valid for multiple value input.
jsexception.unknown.permission.recipient.type=Permission recipient type {0} is not recognized.
jsexception.unknown.property.filter.operation=Property filter operation {0} is not recognized.
jsexception.unknown.timezone=The time zone {0} is not recognized. 
jsexception.uri.not.found=URI {0} cannot be found in the repository.
jsexception.user.credentials.are.expired=User {0}'s credentials have expired.
jsexception.user.disabled=User {0} is disabled.
jsexception.user.expired=User {0} has expired.
jsexception.user.locked=User {0} is locked.
jsexception.user.not.found=User {0} cannot be found.

jsexception.move.folder.source.uri.ancestor.destination={0} cannot be moved to {1}. The destination is a child of the source.
jsexception.move.folder.to.same.folder=You cannot move folder {0} to the same folder {1}.
jsexception.move.resource.to.same.folder=You cannot move resource {0} to the same folder {1}.
jsexception.move.folder.path.already.exists=You cannot move folder {0} to {1}. The destination folder already contains an object with the same name.
jsexception.move.resource.path.already.exists=You cannot move resource {0} to {1}. The destination already contains an object with the same name.
jsexception.copy.resource.referenced.not.readable=Resource {0} is referenced by another resource but it is not readable.
jsexception.copy.resource.circular.reference=A circular reference was detected for resource {0}.
jsexception.folder.moved=Folder {0} has moved to {1}.

report.scheduling.error.filling.report=The report was not completed. An error occurred while executing it.
report.scheduling.error.exporting.report=The export to {0} was not completed. An error interrupted the process.
report.scheduling.error.saving.to.repository={0} was not saved to the repository. An error interrupted the process.
report.scheduling.error.sending.email.notification=The job error notification was not completed. An error occurred while sending it.
report.scheduling.error.system=The job was not completed. A system error occurred while running it.

exception.report.unrecognized.number.type=Unrecognized numeric type {0}
exception.report.unrecognized.date.type=Unrecognized date/time type {0}
exception.report.unsupported.date.input.value=You cannot convert input value {0} of type {1} to a date/time parameter.
exception.report.unsupported.date.parameter.type=Unsupported date/time type {0}
exception.report.unsupported.input.control.data.type=Input control data type {0} is not supported.

jsexception.view.report.could.not.load=The report unit cannot be found or it cannot be loaded.
jsexception.view.page.does.not.exist=Page {0} not found in report
jsexception.view.report.error=An error occurred during report generation
jsexception.export.unknown.argument=Option {0} is not recognized.

jsexception.mondrian.no.connection.for.resource=There is no Mondrian connection for resource {0}.

jsexception.target.org.not.found=Target organization {0} cannot be found in the repository.
jsexception.template.folder.not.found=The template folder cannot be found in the repository.

jsexception.search.duplicate.label=An object with the label {0} already exists in folder {1}.
jsexception.search.duplicate.name=An object with the name {0} already exists in folder {1}.

report.scheduling.error.saving.data.snapshot.disabled=Saving data snapshots is disabled.
report.scheduling.error.data.snapshot.not.populated=Data snapshot was not populated.
report.scheduling.error.data.snapshot.not.persistable=Data snapshot cannot be saved.
report.scheduling.error.saving.data.snapshot.failed=Error while saving data snapshot.

exception.remote.resource.already.exists=Resource {0} already exists.
exception.remote.resource.not.found=Resource {0} not found.
exception.remote.resource.type.not.supported=Resource {0} type is not supported by this operation.
exception.remote.folder.not.found=Folder {0} not found or it is not a folder.
exception.remote.unexpected.error=Unexpected error.
exception.remote.weak.password=Weak password.
exception.remote.resource.not.ready=Task {0} not finished yet.
exception.remote.no.such.export.process=No export task with id {0}.
exception.remote.no.such.import.process=No import task with id {0}.
exception.remote.access.denied=Access denied.
exception.remote.modification.not.allowed=Modification of the resource {0} is not allowed.
exception.remote.illegal.parameter.value.error=The value "{1}" for parameter "{0}" is invalid.
exception.remote.mandatory.parameter.error=The mandatory parameter "{0}" is missing.
exception.remote.import.failed=Import failed. Reason: {0}
exception.remote.import.failed.zip.error=Import failed. Reason: cannot read zip file.
exception.remote.import.failed.content.error=Import failed. Reason: provided zip file is not valid JasperReports Server export file.
exception.remote.resource.querying.not.supported=Resource {0} doesn't support querying
exception.remote.version.not.match=Update failed: versions not match

#AWS data source specific exception messages
aws.exception.datasource.recovery.base=Exception occurred during AWS data source recovery: {0}.
aws.exception.datasource.recovery.timeout=There may be a problem with the JDBC URL, AWS Security Settings or the AWS Data Source recovery process, which can take up to 2 minutes.
aws.exception.datasource.recovery.instance.not.active=The AWS data source instance is not active.
aws.exception.datasource.recovery.is.not.possible=The AWS data source can not be recovered automatically, because your JasperReports Server and database instance are not located in the same VPC.
aws.exception.datasource.recovery.public.ip.not.provided=You must provide the 'Ingress Public IP' of this server in AWS Settings to use automated AWS Data Source setup and recovery.
aws.exception.datasource.recovery.public.ip.not.determined=Unable to determine instance IP address for AWS Data Source recovery.
aws.exception.datasource.load.default.credentials=This instance cannot use <b>EC2 instance credentials</b> to connect automatically to Amazon RDS and Amazon Redshift, because it does not have an Identity and Access Management (IAM) Role assigned. Please read <a class='launcher inline' href='http://community.jaspersoft.com/jaspersoft-aws/launch' target='_blank'>Launching Jaspersoft for AWS</a> to learn how to launch a server which is able to automatically connect to AWS data sources securely.<br><br>As a workaround you can use <b>AWS credentials</b>.
aws.exception.datasource.rds.vpc.recovery=Unable to automatically assign the VPC security group named {0} to RDS instance {1}. The operation is not currently supported by the AWS API. Please manually assign the security group to your RDS instance.
aws.exception.datasource.accessKey.invalid=AWS Access Key is invalid.
aws.exception.datasource.secretKey.invalid=AWS Secret Key is invalid.

#Cluster specific exception messages
cluster.exception.session.attribute.missing=An unrelated error has occurred on the server. Your session was restarted, but any unsaved work has been lost. Please excuse the inconvenience.<p><p>Click the following link to go back to the home page and start again: <a class="launcher inline" href="home.html">Go to home page</a>. If you encounter another error, please contact your administrator.
cluster.exception.session.attribute.missing.popup=An unrelated error has occurred on the server. Your session was restarted, but any unsaved work has been lost. Please excuse the inconvenience.<p><p>Click the button below to go back to the home page and start again. If you encounter another error, please contact your administrator.